driver-mapping/src/main/java/com/datastax/driver/mapping/CommonNamingConventions.java

```diff
@@ -0,0 +1,358 @@
+     * <li>Underscore - @link "http://stackoverflow.com/questions/1899683/is-there-a-standard-in-java-for-underscore-in-front-of-variable-or-class-nam"</li>
+     * </ul>
+     * Those prefixes can be supported so that for example if java convention is
+     * set to UpperCamelCase with `ignorablePrefixes` set to "_" and cassandra
+     * convention is set to LowerSnakeCase, then a field named "_MyXmlParser" will be
+     * translated to "my_xml_parser".
+     */
+    public static class UpperCamelCase extends CamelCase {
+
+        private final boolean upperCaseAbbreviations;
+
+        /**
+         * @param upperCaseAbbreviations {@code true} to uppercase all abbreviations,
+         *                               {@code false} to treat abbreviations as any other word
+         * @param ignorablePrefixes      string prefixes to trim if constant field name prefixes are used
+         */
+        public UpperCamelCase(boolean upperCaseAbbreviations, String... ignorablePrefixes) {
+            super(ignorablePrefixes);
+            this.upperCaseAbbreviations = upperCaseAbbreviations;
+        }
+
+        /**
+         * @param ignorablePrefixes string prefixes to trim if constant field name prefixes are used
+         */
+        public UpperCamelCase(String... ignorablePrefixes) {
+            this(false, ignorablePrefixes);
+        }
+
+        /**
+         * @param upperCaseAbbreviations {@code true} to uppercase all abbreviations,
+         *                               {@code false} to treat abbreviations as any other word
+         */
+        public UpperCamelCase(boolean upperCaseAbbreviations) {
+            this(upperCaseAbbreviations, new String[0]);
+        }
+
+        @Override
+        public String join(List<Word> input) {
+            StringBuilder builder = new StringBuilder();
+            for (Word word : input) {
+                String value;
+                if (upperCaseAbbreviations && word.isAbbreviation()) {
+                    value = word.getValue().toUpperCase();
+                } else {
+                    value = word.getValue().substring(0, 1).toUpperCase() + word.getValue().substring(1).toLowerCase();
+                }
+                builder.append(value);
+            }
+            return builder.toString();
+        }
+
+        @Override
+        public boolean equals(NamingConvention other) {
+            return other instanceof UpperCamelCase &&
+                    ((UpperCamelCase) other).upperCaseAbbreviations == upperCaseAbbreviations &&
+                    super.equals(other);
+        }
+
+    }
+
+    /**
+     * @link "https://en.wikipedia.org/wiki/Snake_case"
+     * <p>
+     * Represents snake case naming convention, meaning all letters are lower cased,
+     * and words are separated by an underscore ("_"). E.g. "my_xml_parser".
+     */
+    public static class LowerSnakeCase extends CharDelimitedNamingConvention {
+
+        public LowerSnakeCase() {
```

Comment:
Would be nice for there to be a static member for this instead of a custom class since there's only one way to configure it.